lib/framework/NetworkSettingsService.cpp

```diff
@@ -364,27 +366,24 @@ void NetworkSettingsService::WiFiEvent(WiFiEvent_t event, WiFiEventInfo_t info)
 #endif
         break;
 
-        // IPv6 specific
+        // IPv6 specific - WiFi
     case ARDUINO_EVENT_WIFI_STA_GOT_IP6:
 #if !TASMOTA_SDK && ESP_IDF_VERSION_MAJOR < 5
         emsesp::EMSESP::logger().info("Local IPv6 (WiFi)=%s", WiFi.localIPv6().toString().c_str());
 #else
-        emsesp::EMSESP::logger().info("Local IPv6=%s", IPAddress(IPv6, (uint8_t *)info.got_ip6.ip6_info.ip.addr, 0).toString().c_str());
+        emsesp::EMSESP::logger().info("IPv6 (WiFi)=%s", IPAddress(IPv6, (uint8_t *)info.got_ip6.ip6_info.ip.addr, 0).toString().c_str());
 #endif
         emsesp::EMSESP::system_.has_ipv6(true);
         break;
 
-        // IPv6 specific
-        // This a bug in arduino where this is triggered twice, so we prevent it
+        // IPv6 specific - Eth
     case ARDUINO_EVENT_ETH_GOT_IP6:
-        if (!emsesp::EMSESP::system_.has_ipv6()) {
 #if !TASMOTA_SDK && ESP_IDF_VERSION_MAJOR < 5
-            emsesp::EMSESP::logger().info("Local IPv6 (Eth)=%s", ETH.localIPv6().toString().c_str());
+        emsesp::EMSESP::logger().info("Local IPv6 (Eth)=%s", ETH.localIPv6().toString().c_str());
 #else
-            emsesp::EMSESP::logger().info("Local IPv6=%s", IPAddress(IPv6, (uint8_t *)info.got_ip6.ip6_info.ip.addr, 0).toString().c_str());
+        emsesp::EMSESP::logger().info("IPv6 (Eth)=%s", IPAddress(IPv6, (uint8_t *)info.got_ip6.ip6_info.ip.addr, 0).toString().c_str());
 #endif
-            emsesp::EMSESP::system_.has_ipv6(true);
-        }
+        emsesp::EMSESP::system_.has_ipv6(true);
         break;
```

Comment:
I've tested how to show the different types of ip6 addresses also for arduino 2.xx, try this:
```
    case ARDUINO_EVENT_ETH_GOT_IP6: {
#if !TASMOTA_SDK && ESP_IDF_VERSION_MAJOR < 5
        auto ip6 = IPv6Address((uint8_t *)info.got_ip6.ip6_info.ip.addr).toString();
#else
        auto ip6 = IPAddress(IPv6, (uint8_t *)info.got_ip6.ip6_info.ip.addr, 0).toString();
#endif
        if (ip6.startsWith("fe80")) {
            emsesp::EMSESP::logger().info("IPv6 local: %s", ip6.c_str());
        } else if (ip6.startsWith("fd")) {
            emsesp::EMSESP::logger().info("IPv6 ULA: %s", ip6.c_str());
        } else {
            emsesp::EMSESP::logger().info("IPv6 global: %s", ip6.c_str());
        }
        emsesp::EMSESP::system_.has_ipv6(true);
    } break;
```

Reply:
nice, i'll add.